include/exadg/incompressible_navier_stokes/spatial_discretization/generalized_newtonian_model.cpp

```diff
@@ -0,0 +1,334 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2023 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#include <exadg/incompressible_navier_stokes/spatial_discretization/generalized_newtonian_model.h>
+
+namespace ExaDG
+{
+namespace IncNS
+{
+template<int dim, typename Number>
+GeneralizedNewtonianModel<dim, Number>::GeneralizedNewtonianModel()
+  : Base(), generalized_newtonian_model(GeneralizedNewtonianViscosityModel::Undefined)
+{
+}
+
+template<int dim, typename Number>
+GeneralizedNewtonianModel<dim, Number>::~GeneralizedNewtonianModel()
+{
+}
+
+template<int dim, typename Number>
+void
+GeneralizedNewtonianModel<dim, Number>::initialize(
+  dealii::MatrixFree<dim, Number> const &                matrix_free_in,
+  std::shared_ptr<Operators::ViscousKernel<dim, Number>> viscous_kernel_in,
+  GeneralizedNewtonianModelData const &                  generalized_newtonian_model_data_in,
+  unsigned int                                           dof_index_velocity_in,
+  unsigned int                                           quad_index_velocity_linear_in)
```

Comment:
I'd also make these parameters `const`. Sorry about commenting so much on `const`ness but it is very important 😆.

Reply:
The parameters are passed by reference, so `const` does not really have an effect, but is rather a help in reading the code. I agree to write `const` in such cases for clarity.

Generally, I think we should be strict about constness in all the include/exadg files. I think we can treat constness somewhat shabbily in the application.h files, since no other code depends on that. Ideally, we would of course care about constness everywhere.

Reply:
You are right that `const` does not have an effect on the variable that was passed into this function, but it does on the variable that is local in this  scope. One would still interpret this as a function that modifies the local variable, which would still be a significant misunderstanding in my opinion and be no different than declaring a non-const local variable.